Added functionality for client browsing of queues based on JMS QueueB…

wxlund opened the pull request:
…rowser.  It

is completed separate from the other functionality so should merge but I'm doing something
wrong with the runner that it thinks I'm trying to find the "send" profile instead of "browse".

wxlund committed:
Added functionality for client browsing of queues based on JMS QueueBrowser.  It
is completed separate from the other functionality so should merge but I'm doing something
wrong with the runner that it thinks I'm trying to find the "send" profile instead of "browse".

wxlund committed:
Merge https://github.com/sdeeg-pivotal/rabbitmq-jms-samples into add-browse

wxlund commented:
Is there something obvious you can see that's preventing my new "browse" profile not to pick up the @Bean for QueueBrowser?

sdeeg-pivotal commented:
There are a couple of problems with it.

0) Get my latest code

1) Annotate BrowseConfig with @Configuration

2) There are two session beans now, so you have to disambiguate which one is passed to queueBrowser(...).  Change the param name to jmsSession.  (The second session is being use for returning messages outside of the session being used for the transactions in the poison message pattern.)

3) I don't understand @Qualifier, but when I took it out (and did above) the app runs

wxlund committed:
Applied suggestions.  QueueBrowse is working now although there is an issue with passing in the exchange.
The default exchange works fine.

wxlund committed:
Merge https://github.com/sdeeg-pivotal/rabbitmq-jms-samples into add-browse

wxlund committed:
Somewhat fixed queuebrowser.  Following scenario works:

--jms.persistent=true  --nummessages=100 --spring.profiles.active=browse   --amqp.uri=amqp://guest:guest@localhost:5672/%2f

wxlund commented:
I’m 5 commits ahead of you now so code is current.  Can’t believe I forgot @Configuration. I disambiguated by naming mine queueSession. I think @Qualifier is another way to disambiguate but I must not have used it correctly.  The happy path is now working and ready to merge.

> On Feb 16, 2016, at 8:27 PM, Scott Deeg <notifications@github.com <mailto:notifications@github.com>> wrote:
> 
> There are a couple of problems with it.
> 
> 0) Get my latest code
> 
> 1) Annotate BrowseConfig with @Configuration https://github.com/Configuration
> 2) There are two session beans now, so you have to disambiguate which one is passed to queueBrowser(...). Change the param name to jmsSession. (The second session is being use for returning messages outside of the session being used for the transactions in the poison message pattern.)
> 
> 3) I don't understand @Qualifier https://github.com/Qualifier, but when I took it out (and did above) the app runs
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/sdeeg-pivotal/rabbitmq-jms-samples/pull/2#issuecomment-185014991.
